# Inferential Statistics - Temporal Trends 

## Purpose
This notebook evaluates whether key business outcomes have changed over time using simple statistical models.

Specifically, the analysis tests whether:

1. Profitability changed after 2013.
2. Sales exhibit a long-run growth or decline trend.
3. Return risk is increasing or decreasing over time.
4. Discount intensity has changed over time.

These models complement the descriptive temporal analysis by determining whether observed patterns reflect statistically meaningful changes.

## Contents

- [1. Setup and Dta Prepartion](#1-setup-and-data-preparation)
- [2. Pre/Post 2013 Performance](#2-prepost-2013-performance)
- [3. Year-over-Year Growth](#3-year-over-year-growth)
- [4. Return Rate Over Time](#4-return-rate-over-time)
- [5. Discount Intensity Over Time](#5-discount-intensity-over-time)
- [6. Interpretation](#6-interpretation)

## 1. Setup and Data Preparation

In [2]:
# Load required libraries
library(tidyverse)
library(janitor)
library(dplyr)
library(ggplot2)
library(skimr)
library(purrr)
library(lubridate)

# Source helper scripts
source("../../R/apply_factors.R")
source("../../R/analysis_helpers.R")
source("../../R/temporal_helpers.R")

# Load data
tables <- list(
  Orders  = readr::read_csv("../../data/processed/Orders.csv"),
  Returns = readr::read_csv("../../data/processed/Returns.csv"),
  People  = readr::read_csv("../../data/processed/People.csv")
)

# Apply factor transformations
tables <- apply_factors(tables)

# Extract tables
orders  <- tables$Orders
returns <- tables$Returns
people  <- tables$People

Rows: 51290 Columns: 21
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (13): order_id, ship_mode, customer_name, segment, state, country, mark...
dbl   (6): sales, quantity, discount, profit, shipping_cost, year
date  (2): order_date, ship_date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1173 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): order_id, market
lgl (1): returned

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 14 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): person, region

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the co

## 2. Pre/Post 2013 Performance

**Observed**

Sales and profit appear to exhibit a step change around 2013.

**Inference Question**

> Was there a statistically significant change in profitability after 2013?

**Method**

Regression with a post-2013 indicator:
- Outcome: profit
- Predictor: post_2013

This framing treats 2013 as a quasi-experimental breakpoint.

In [3]:
orders <- orders |>  
    mutate(    
        year = lubridate::year(order_date),    
        post_2013 = as.integer(year >= 2013)  
    )

post2013_lm <- lm(
    profit ~ post_2013,
    data = orders
)

summary(post2013_lm)


Call:
lm(formula = profit ~ post_2013, data = orders)

Residuals:
    Min      1Q  Median      3Q     Max 
-6629.1   -28.9   -19.4     8.2  8370.8 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   27.874      1.235  22.577   <2e-16 ***
post_2013      1.258      1.580   0.796    0.426    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 174.4 on 51288 degrees of freedom
Multiple R-squared:  1.236e-05,	Adjusted R-squared:  -7.14e-06 
F-statistic: 0.6338 on 1 and 51288 DF,  p-value: 0.426


## 3. Year-over-Year Growth

**Observed**

Some years exhibit stronger sales growth than others.

**Inference Question**

> Is long-run sales growth accelerating or decelerating over time?

**Method**

Trend regression on log-transformed sales:
- Outcome: log(sales)
- Predictor: year

This model captures the average long-run growth trajectory.

In [4]:
sales_trend_lm <- lm(
    log1p(sales) ~ year,
    data = orders
)
summary(sales_trend_lm)


Call:
lm(formula = log1p(sales) ~ year, data = orders)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1430 -1.0540 -0.0568  1.0180  5.5135 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)
(Intercept)  6.890273  11.515596   0.598    0.550
year        -0.001182   0.005721  -0.207    0.836

Residual standard error: 1.424 on 51288 degrees of freedom
Multiple R-squared:  8.317e-07,	Adjusted R-squared:  -1.867e-05 
F-statistic: 0.04266 on 1 and 51288 DF,  p-value: 0.8364


## 4. Return Rate Over Time

**Observed**

Return rates vary across years.

**Inference Question**

> Is return risk increasing or decreasing over time?

**Method**

Logistic regression:
- Binary outcome: returned
- Predictor: year

In [5]:
orders <- orders |>  
    left_join(    
        returns |>      
            mutate(returned = 1L),    
        by = "order_id",    
        relationship = "many-to-many"  
    ) |>  
    mutate(    
        returned = ifelse(is.na(returned), 0L, returned)  
    )

return_time_logit <- glm(
    returned ~ year,
    data = orders,
    family = binomial(link = "logit")
)

summary(return_time_logit)


Call:
glm(formula = returned ~ year, family = binomial(link = "logit"), 
    data = orders)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept) 57.70735   34.01044   1.697   0.0897 .
year        -0.03004    0.01690  -1.778   0.0754 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 23159  on 51294  degrees of freedom
Residual deviance: 23156  on 51293  degrees of freedom
AIC: 23160

Number of Fisher Scoring iterations: 5


## 5. Discount Intensity Over Time

**Observed**

Discount usage appears to increase over time.

**Inference Question**

> Is discount intensity trending upward over time?

**Method**

Linear regression:
- Outcome: discount
- Predictor: year

In [6]:
discount_trend_lm <- lm(
    discount ~ year,
    data = orders
)
summary(discount_trend_lm)


Call:
lm(formula = discount ~ year, data = orders)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.14493 -0.14265 -0.14151  0.05849  0.70621 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)
(Intercept)  2.4387829  1.7166278   1.421    0.155
year        -0.0011407  0.0008529  -1.337    0.181

Residual standard error: 0.2123 on 51293 degrees of freedom
Multiple R-squared:  3.487e-05,	Adjusted R-squared:  1.538e-05 
F-statistic: 1.789 on 1 and 51293 DF,  p-value: 0.1811


## 6. Interpretation

### Profitability change after 2013
The regression testing whether profit changed after 2013 shows that the **post_2013 coefficient is not statistically significant (p = 0.426)**.

This indicates there is **no evidence of a structural shift in average order profitability after 2013**.  
Despite descriptive patterns suggesting a possible change, the statistical test shows that profit levels before and after 2013 are not meaningfully different.

---

### Long-run sales growth
The trend regression of log-transformed sales on year produces a **non-significant year coefficient (p = 0.836)** with an extremely small R².

This indicates **no detectable long-run upward or downward sales trend** at the order level over the observed period.  
Year-to-year variation in sales likely reflects short-term fluctuations rather than a persistent growth or decline trend.

---

### Return rate over time
The logistic regression modeling return probability as a function of year produces a **negative coefficient on year (−0.030)** with **p ≈ 0.075**.

While not statistically significant at the conventional 5% level, this result suggests a **weak downward trend in return probability over time**, meaning return risk may have decreased slightly across the dataset.

---

### Discount intensity over time
The regression of discount on year shows a **small negative coefficient with p = 0.182**, indicating **no statistically significant change in average discount levels over time**.

This suggests that the firm’s discounting behavior has remained relatively stable across the years analyzed.

---

### Overall conclusion
Across all models, the temporal analysis provides **little evidence of strong structural changes over time**. Profitability, sales levels, return risk, and discount intensity appear broadly stable across years.

This indicates that most variation in performance likely arises from **cross-sectional factors** such as product mix, region, shipping strategy, or discount policy rather than from long-term temporal trends.